symengine/pow.cpp

```diff
@@ -161,6 +163,14 @@ RCP<const Basic> pow(const RCP<const Basic> &a, const RCP<const Basic> &b)
                 .pow(*rcp_static_cast<const Number>(b));
         }
     }
+    if (is_a<RealDouble>(*a) and is_a<Constant>(*b)) {
```

Comment:
Surely this should be a `Number` instead of a `RealDouble`. This would cater for the case where `a` is complex, derives from `NumberWrapper` etc. If so, then I'm guessing that one should likely call `a.pow(eval_double(*b))` or similar.